pkg/cnab/provider/parameters.go

```diff
@@ -37,31 +37,39 @@ func (d *Runtime) loadParameters(claim *claim.Claim, rawOverrides map[string]str
 			return nil, errors.Wrapf(err, "unable to convert parameter's %s value %s to the destination parameter type %s", key, rawValue, def.Type)
 		}
 
-		// If this parameter applies to the current action, set the override accordingly
-		if param.AppliesTo(action) {
-			overrides[key] = value
-		} else {
-			// Otherwise, set to current parameter value on the claim, if exists
+		overrides[key] = value
+		// If this parameter does not apply to the current action, defer to the claim value, if exists
+		if !param.AppliesTo(action) {
 			if _, exists := claim.Parameters[key]; exists {
 				overrides[key] = claim.Parameters[key]
 			}
-			if d.Debug {
-				fmt.Fprintf(d.Err,
-					"override supplied for '%s', but this parameter is not configured to apply for action '%s'; skipping\n",
-					key, action)
-			}
 		}
 	}
 
-	// rawOverrides may supply no entry for a parameter designated as required
-	// *but* should not apply to this action.
-	// When this occurs, we set an override to the current value in the claim such that
-	// bundle.ValuesOrDefaults does not return an error
+	// rawOverrides (meaning, user-supplied overrides at time of action invocation)
+	// may supply no entry for a parameter designated as required *but* that does not apply to this action.
+	//
+	// When this occurs, we set an override to either the current value in the claim or the default value.
+	// If neither exists, the zero value according to the parameter type will be used.
+	// Otherwise, if unset/nil, json validation in bundle.ValuesOrDefaults would return an error
 	for name, param := range bun.Parameters {
+		def, ok := bun.Definitions[param.Definition]
+		if !ok {
+			return nil, fmt.Errorf("definition %s not defined in bundle", param.Definition)
```

Comment:
nit: this error message is probably going to pop up in the middle of an action without enough context to understand that we are referring to a parameter's definition. I suggest saying "parameter definition %s..."